app/controllers/split_checkout_controller.rb

```diff
@@ -25,45 +25,114 @@ def edit
     # a version of paypal that uses this controller, and more specifically
     # the #action_failed method, then we can remove this call
     # OrderCheckoutRestart.new(@order).call
+    @order.errors.clear
+    @order.bill_address.errors.clear
+    @order.ship_address.errors.clear
+    @ship_address_same_as_billing = "1" if ship_address_matches_bill_address?
   rescue Spree::Core::GatewayError => e
     rescue_from_spree_gateway_error(e)
   end
 
   def update
-    if confirm_order || update_order
+    load_shipping_method
+    handle_shipping_method_selection
+
+    confirm_or_update = confirm_order || update_order
+    if confirm_or_update && advance_order_state
       clear_invalid_payments
-      advance_order_state
       redirect_to_step
     else
-      flash.now[:error] = "Saving failed, please update the highlighted fields"
+      if @shipping_method_id.blank?
+        @order.errors.add(:base, "no_shipping_method_selected")
+      end
+      flash.now[:error] = "#{I18n.t('split_checkout.errors.global')}"
```

Comment:
```suggestion
      flash.now[:error] = I18n.t('split_checkout.errors.global')
```

I thought there was a rubocop rule for this...

Reply:
I think I already fixed it in a later commit :+1: